wrappers/obj-c/ODWPrivacyGuard.mm

```diff
@@ -98,6 +98,10 @@ +(void)initializePrivacyGuard:(ILogger *)logger withODWPrivacyGuardInitConfig:(O
     {
         config.SemanticContextNotificationEventName = [[initConfigObject semanticContextNotificationEventName] UTF8String];
     }
+    if ([initConfigObject metadataProvider] != nill)
+    {
+        config.metadataProvider = [[]]
+    }
```

Comment:
`config.metadataProvider` should be `config.MetadataProvider` defined [here](https://github.com/microsoft/cpp_client_telemetry_modules/blob/8539da62dd5e41e2847337df1d95f0f841d3a738/privacyguard/InitializationConfiguration.hpp). However, current implementation does not assign metadata provider to underlying data structure in PG C++. This is because in PG C++, Metadata provider is defined as 
`const std::shared_ptr<PrivacyConcernMetadataProvider> MetadataProvider;`, so you will need to convert your `ODWPrivacyConcernMetadataProvider` to this type. Unfortunately, there is no direct conversion between Obj-C and C++ class implementations. One way I found to achieve this is to use composition. In this [link](https://stackoverflow.com/a/9741994), it talks about how to do this.

I think this is what you need 
`
// h file
@interface ODWPrivacyConcernMetadataProvider: NSObject
{
    @public
    PrivacyConcernMetadataProvider *cppObject;
}

- (NSString *)getDatabaseNameForRecord:(id)record;
- // rest of interface functions

@end

// mm file
@implementation MyCPPWrapper

- (NSString *)getDatabaseNameForRecord:(id)record;
{
    if (cppObject)
        cppObject->GetDatabaseName(id);
}

@end
`